Library/Homebrew/requirement.rb

```diff
@@ -109,7 +109,7 @@ def env_proc
   end
 
   def ==(other)
-    instance_of?(other.class) && name == other.name && tags == other.tags
+    instance_of?(other.class) && hash == other.hash
```

Comment:
The test is now expecting `eql?` to call `hash`, which it wasn't before. Anyway, I think it would be better to move the `hash` check to a separate test, so ignore this.